crates/hir_def/src/type_ref.rs

```diff
@@ -79,7 +80,7 @@ pub enum TypeRef {
     Path(Path),
     RawPtr(Box<TypeRef>, Mutability),
     Reference(Box<TypeRef>, Option<LifetimeRef>, Mutability),
-    Array(Box<TypeRef> /*, Expr*/),
+    Array(Box<TypeRef>, ConstScalar),
```

Comment:
well, this is quite hacky. I guess it's fine for now since it doesn't prevent us from making progress, but obviously we really should be lowering the array size expression properly. Can you add a FIXME comment that makes clear that this is a temporary solution?

Reply:
Added.